controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -113,6 +115,17 @@ func (r *AutoscalingRunnerSetReconciler) Reconcile(ctx context.Context, req ctrl
 			return ctrl.Result{}, err
 		}
 
+		requeue, err := r.removeFinalizersFromPermissionResources(ctx, autoscalingRunnerSet, log)
+		if err != nil {
+			log.Error(err, "Failed to clean up permissions")
+			return ctrl.Result{}, nil
+		}
+
+		if requeue {
+			log.Info("Waiting for role/rolebinding/serviceaccount to be deleted")
```

Comment:
```suggestion
			log.Info("Waiting for dependent resources to be deleted")
```

if you change the method name. 😄